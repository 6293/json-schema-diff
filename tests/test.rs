use insta::{assert_debug_snapshot, glob, with_settings};
use json_schema_diff::diff;
use serde_json::Value;

#[test]
fn test_from_fixtures() {
    let test = |path: &std::path::Path| {
        let contents = std::fs::read_to_string(path).unwrap();
        let values: Value = serde_json::from_str(&contents).unwrap();
        let diff = diff(values["lhs"].clone(), values["rhs"].clone()).unwrap();
        with_settings!({ info => &values }, {
            assert_debug_snapshot!(diff);
        });
    };
    glob!("../tests/fixtures", "**/*.json", test);
}
