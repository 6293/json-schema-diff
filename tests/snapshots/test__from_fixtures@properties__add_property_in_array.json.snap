---
source: tests/test.rs
expression: diff
info:
  lhs:
    definitions:
      Hello:
        type: object
    items:
      - const: start_unmerge
      - $ref: "#/definitions/Hello"
    type: array
  rhs:
    definitions:
      Hello:
        properties:
          transaction_id:
            type: string
        type: object
    items:
      - const: start_unmerge
      - $ref: "#/definitions/Hello"
    type: array
input_file: tests/fixtures/properties/add_property_in_array.json
---
[
    Change {
        path: ".1",
        change: PropertyAdd {
            lhs_additional_properties: true,
            added: "transaction_id",
        },
    },
]
